pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -199,7 +200,7 @@ func (p *ConfigurationEditFlags) Apply(
 
 	template := &service.Spec.Template
 
-	err := p.PodSpecFlags.ResolvePodSpec(&template.Spec.PodSpec, cmd.Flags())
+	err := p.PodSpecFlags.ResolvePodSpec(&template.Spec.PodSpec, cmd.Flags(), os.Args)
```

Comment:
`os.Args` -> this is the only way how to get all orderd list of all arguments passed to the command (so we do the right order of `--env` and `--env-value-from`. Unfortunately Cobra can't do that, it works on already processed args and flags.